--- a/src/sbin/dmesg/dmesg.c
+++ b/src/sbin/dmesg/dmesg.c
@@ -87,7 +87,7 @@
 	clear = false;
 	(void) setlocale(LC_CTYPE, "");
 	memf = nlistf = NULL;
-	while ((ch = getopt(argc, argv, "acM:N:")) != -1)
+	while ((ch = getopt(argc, argv, "acM:N:s:n:")) != -1)
 		switch(ch) {
 		case 'a':
 			all = true;
@@ -101,6 +101,9 @@
 		case 'N':
 			nlistf = optarg;
 			break;
+		case 's':
+		case 'n':
+			break;
 		case '?':
 		default:
 			usage();
--- a/src/sbin/dmesg/Makefile
+++ b/src/sbin/dmesg/Makefile
@@ -7,4 +7,6 @@
 LDADD=	-lkvm
 DPADD=	${LIBKVM}
 
+LDADD+=	-lbsd
+
 .include <bsd.prog.mk>
