--- a/sbin/ifconfig/Makefile
+++ b/sbin/ifconfig/Makefile
@@ -27,20 +27,17 @@
 .endif
 
 SRCS+=	ifclone.c		# clone device support
-SRCS+=	ifmac.c			# MAC support
 SRCS+=	ifmedia.c		# SIOC[GS]IFMEDIA support
-SRCS+=	iffib.c			# non-default FIB support
 SRCS+=	ifvlan.c		# SIOC[GS]ETVLAN support
 SRCS+=	ifgre.c			# GRE keys etc
 SRCS+=	ifgif.c			# GIF reversed header workaround
 
 SRCS+=	ifieee80211.c regdomain.c # SIOC[GS]IEEE80211 support
 DPADD+=	${LIBBSDXML} ${LIBSBUF}
-LDADD+=	-lbsdxml -lsbuf
+LDADD+=	-lexpat -lsbuf -lfreebsd-glue
 
 SRCS+=	ifcarp.c		# SIOC[GS]VH support
 SRCS+=	ifgroup.c		# ...
-SRCS+=	ifpfsync.c		# pfsync(4) support
 
 SRCS+=	ifbridge.c		# bridge support
 SRCS+=	iflagg.c		# lagg support
@@ -65,6 +62,7 @@
 MAN=	ifconfig.8
 
 CFLAGS+= -Wall -Wmissing-prototypes -Wcast-qual -Wwrite-strings -Wnested-externs
+LDADD+= -lbsd ../../lib/linkaddr.o
 WARNS?=	2
 
 .include <bsd.prog.mk>
--- a/sbin/ifconfig/ifconfig.c
+++ b/sbin/ifconfig/ifconfig.c
@@ -76,6 +76,13 @@
 
 #include "ifconfig.h"
 
+#ifndef IFCAP_RXCSUM_IPV6
+#define IFCAP_RXCSUM_IPV6 0x200000
+#endif
+#ifndef IFCAP_TXCSUM_IPV6
+#define IFCAP_TXCSUM_IPV6 0x400000
+#endif
+
 /*
  * Since "struct ifreq" is composed of various union members, callers
  * should pay special attention to interprete the value.
@@ -158,7 +165,7 @@
 	all = downonly = uponly = namesonly = noload = verbose = 0;
 
 	/* Parse leading line options */
-	strlcpy(options, "adklmnuv", sizeof(options));
+	strlcpy(options, "+adklmnuv", sizeof(options));
 	for (p = opts; p != NULL; p = p->next)
 		strlcat(options, p->opt, sizeof(options));
 	while ((c = getopt(argc, argv, options)) != -1) {
@@ -954,7 +961,7 @@
 	putchar('\n');
 
 	for (;;) {
-		if ((descr = reallocf(descr, descrlen)) != NULL) {
+		if ((descr = realloc(descr, descrlen)) != NULL) {
 			ifr.ifr_buffer.buffer = descr;
 			ifr.ifr_buffer.length = descrlen;
 			if (ioctl(s, SIOCGIFDESCR, &ifr) == 0) {
--- a/sbin/ifconfig/af_link.c
+++ b/sbin/ifconfig/af_link.c
@@ -47,6 +47,8 @@
 #include <net/if_types.h>
 #include <net/ethernet.h>
 
+#include "../../lib/linkaddr.h"
+
 #include "ifconfig.h"
 
 static struct ifreq link_ridreq;
--- a/sbin/ifconfig/ifieee80211.c
+++ b/sbin/ifconfig/ifieee80211.c
@@ -83,6 +83,9 @@
 #include <fcntl.h>
 #include <inttypes.h>
 #include <stdio.h>
+
+
+#include "../../lib/linkaddr.h"
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
