--- a/src/sbin/mount/mount.c
+++ b/src/sbin/mount/mount.c
@@ -672,11 +672,13 @@
 			(void)printf(", reads: sync %ju async %ju",
 			    (uintmax_t)sfp->f_syncreads,
 			    (uintmax_t)sfp->f_asyncreads);
+#ifdef __FreeBSD__
 		if (sfp->f_fsid.val[0] != 0 || sfp->f_fsid.val[1] != 0) {
 			printf(", fsid ");
 			for (i = 0; i < sizeof(sfp->f_fsid); i++)
 				printf("%02x", ((u_char *)&sfp->f_fsid)[i]);
 		}
+#endif
 	}
 	(void)printf(")\n");
 }
