--- a/src/sbin/route/route.c
+++ b/src/sbin/route/route.c
@@ -167,7 +167,7 @@
 	if (argc < 2)
 		usage(NULL);
 
-	while ((ch = getopt(argc, argv, "nqdtv")) != -1)
+	while ((ch = getopt(argc, argv, "+nqdtv")) != -1)
 		switch(ch) {
 		case 'n':
 			nflag = 1;
@@ -247,6 +249,11 @@
 
 	if (fib < 0)
 		return (0);
+
+#ifndef SO_SETFIB
+#define SO_SETFIB 0x1014
+#endif
+
 	return (setsockopt(s, SOL_SOCKET, SO_SETFIB, (void *)&fib,
 	    sizeof(fib)));
 }
