--- a/src/sbin/swapon/swapon.c
+++ b/src/sbin/swapon/swapon.c
@@ -87,6 +87,8 @@
 	int sflag, lflag, late, hflag;
 	const char *etc_fstab;
 
+	setprogname (argv[0]);
+
 	sflag = lflag = late = hflag = 0;
 	if ((ptr = strrchr(argv[0], '/')) == NULL)
 		ptr = argv[0];
@@ -98,7 +100,7 @@
 	
 	doall = 0;
 	etc_fstab = NULL;
-	while ((ch = getopt(argc, argv, "AadghklLmqsUF:")) != -1) {
+	while ((ch = getopt(argc, argv, "AadghklLmqsUF:ve")) != -1) {
 		switch(ch) {
 		case 'A':
 			if (which_prog == SWAPCTL) {
@@ -154,6 +156,9 @@
 		case 'F':
 			etc_fstab = optarg;
 			break;
+		case 'v':
+		case 'e':
+			break;
 		case '?':
 		default:
 			usage();
