---
 sbin/sysctl/sysctl.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/src/sbin/sysctl/sysctl.c
+++ b/src/sbin/sysctl/sysctl.c
@@ -58,6 +58,7 @@
 #include <string.h>
 #include <sysexits.h>
 #include <unistd.h>
+#include <time.h>
 
 static const char *conffile;
 
@@ -471,6 +472,7 @@
 static int
 S_loadavg(int l2, void *p)
 {
+#if 0
 	struct loadavg *tv = (struct loadavg*)p;
 
 	if (l2 != sizeof(*tv)) {
@@ -481,6 +483,7 @@
 		(double)tv->ldavg[0]/(double)tv->fscale,
 		(double)tv->ldavg[1]/(double)tv->fscale,
 		(double)tv->ldavg[2]/(double)tv->fscale);
+#endif
 	return (0);
 }
 
