--- a/usr.bin/netstat/netgraph.c
+++ b/usr.bin/netstat/netgraph.c
@@ -76,8 +76,8 @@
 		const char *const modname = "ng_socket.ko";
 /* XXX We should get "mpath" from "sysctl kern.module_path" */
 		const char *mpath[] = { "/", "/boot/", "/modules/", NULL };
-		struct nlist sym[] = { { .n_name = "_ngsocklist" },
-				       { .n_name = NULL } };
+		struct nlist sym[] = { {{ .n_name = "_ngsocklist" }},
+				       {{ .n_name = NULL }} };
 		const char **pre;
 		struct kld_file_stat ks;
 		int fileid;
--- a/usr.bin/netstat/netstat.h
+++ b/usr.bin/netstat/netstat.h
@@ -32,6 +32,19 @@
 
 #include <sys/cdefs.h>
 
+
+
+
+
+
+/* Only used internally, so can be outside the range of valid IP protocols. */
+#define IPPROTO_DIVERT          258             /* divert pseudo-protocol */
+
+
+
+
+
+
 extern int	Aflag;	/* show addresses of protocol control block */
 extern int	aflag;	/* show all sockets (including servers) */
 extern int	bflag;	/* show i/f total bytes in/out */
--- a/usr.bin/netstat/Makefile
+++ b/usr.bin/netstat/Makefile
@@ -30,6 +30,7 @@
 BINMODE=2555
 DPADD=	${LIBKVM} ${LIBMEMSTAT} ${LIBUTIL}
 LDADD=	-lkvm -lmemstat -lutil
+LDADD+= -lrt -lbsd -lfreebsd-glue
 
 .if ${MK_NETGRAPH_SUPPORT} != "no"
 SRCS+=	netgraph.c
