--- a/sbin/camcontrol/camcontrol.c
+++ b/sbin/camcontrol/camcontrol.c
@@ -134,12 +134,12 @@
 };
 
 #ifndef MINIMALISTIC
-static const char scsicmd_opts[] = "a:c:dfi:o:r";
-static const char readdefect_opts[] = "f:GP";
-static const char negotiate_opts[] = "acD:M:O:qR:T:UW:";
-static const char smprg_opts[] = "l";
-static const char smppc_opts[] = "a:A:d:lm:M:o:p:s:S:T:";
-static const char smpphylist_opts[] = "lq";
+static const char scsicmd_opts[] = "+a:c:dfi:o:r";
+static const char readdefect_opts[] = "+f:GP";
+static const char negotiate_opts[] = "+acD:M:O:qR:T:UW:";
+static const char smprg_opts[] = "+l";
+static const char smppc_opts[] = "+a:A:d:lm:M:o:p:s:S:T:";
+static const char smpphylist_opts[] = "+lq";
 #endif
 
 struct camcontrol_opts option_table[] = {
--- a/sbin/camcontrol/Makefile
+++ b/sbin/camcontrol/Makefile
@@ -12,7 +12,7 @@
 WARNS?= 3
 .endif
 DPADD=	${LIBCAM} ${LIBSBUF} ${LIBUTIL}
-LDADD=	-lcam -lsbuf -lutil
+LDADD=	-lcam -lsbuf -lutil -lbsd
 MAN=	camcontrol.8
 
 .include <bsd.prog.mk>
