--- a/sbin/geom/misc/subr.c
+++ b/sbin/geom/misc/subr.c
@@ -177,7 +177,7 @@
 	if (*s != '\0')
 		return (EINVAL);
 done:
-	if ((OFF_MAX / unit) < mult || (OFF_MAX / mult / unit) < number)
+	if ((INT64_MAX / unit) < mult || (INT64_MAX / mult / unit) < number)
 		return (ERANGE);
 	number *= mult * unit;
 	if (number % sectorsize)
--- a/sys/crypto/sha2/sha2.c
+++ b/sys/crypto/sha2/sha2.c
@@ -67,8 +67,10 @@
  *
  */
 
-#if defined(__bsdi__) || defined(__FreeBSD__)
+#if defined(_KERNEL) && (defined(__bsdi__) || defined(__FreeBSD__))
 #define assert(x)
+#else
+#include <assert.h>
 #endif
 
 
