Index: freebsd-utils-jessie/src/usr.sbin/acpi/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/acpi/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/acpi/Makefile
@@ -2,6 +2,7 @@
 # $Id: Makefile,v 1.1 2000/07/14 18:16:22 iwasaki Exp $
 # $FreeBSD: stable/10/usr.sbin/acpi/Makefile 118643 2003-08-08 03:20:59Z njl $
 
-SUBDIR= acpiconf acpidb acpidump iasl
+SUBDIR= acpiconf
+# acpidb acpidump iasl
 
 .include <bsd.subdir.mk>
Index: freebsd-utils-jessie/src/sbin/ifconfig/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/ifconfig/Makefile
+++ freebsd-utils-jessie/src/sbin/ifconfig/Makefile
@@ -38,6 +38,8 @@ SRCS+=	ifieee80211.c regdomain.c # SIOC[
 DPADD+=	${LIBBSDXML} ${LIBSBUF}
 LDADD+=	-lbsdxml -lsbuf
 
+LDADD+= -lbsd -lfreebsd-glue
+
 SRCS+=	carp.c			# SIOC[GS]VH support
 SRCS+=	ifgroup.c		# ...
 .if ${MK_PF} != "no"
Index: freebsd-utils-jessie/src/usr.sbin/ppp/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/ppp/Makefile
@@ -38,10 +38,10 @@ BINMODE=550
 BINMODE=4550
 BINOWN=	root
 .endif
-BINGRP=	network
+BINGRP=	dialout
 M4FLAGS=
 
-LDADD= -lcrypt -lmd -lutil -lz
+LDADD= -lcrypt      -lutil-freebsd -lz
 DPADD=	${LIBCRYPT} ${LIBMD} ${LIBUTIL} ${LIBZ}
 
 .if defined(PPP_CONFDIR) && !empty(PPP_CONFDIR)
Index: freebsd-utils-jessie/src/usr.sbin/nfsd/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/nfsd/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/nfsd/Makefile
@@ -3,5 +3,7 @@
 
 PROG=	nfsd
 MAN=	nfsd.8 nfsv4.4 stablerestart.5
+CFLAGS+= -I${.CURDIR}/../../sys
+LDADD=   -ltirpc
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/usr.sbin/rpc.lockd/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/rpc.lockd/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/rpc.lockd/Makefile
@@ -5,23 +5,27 @@ PROG=	rpc.lockd
 MAN=	rpc.lockd.8
 MLINKS=	rpc.lockd.8 lockd.8
 SRCS=	kern.c nlm_prot_svc.c lockd.c lock_proc.c lockd_lock.c
+SRCS+= nlm_prot_xdr.c sm_inter_xdr.c
 
-CFLAGS+= -I. -I${DESTDIR}/usr/include/rpcsvc
+CFLAGS+= -I. -I${.CURDIR}/../../include/rpcsvc
 WARNS?= 3
 
 DPADD=	${LIBRPCSVC} ${LIBUTIL}
-LDADD=	-lrpcsvc -lutil
+LDADD= -ltirpc -lfreebsd-glue
 
 CLEANFILES= nlm_prot_svc.c nlm_prot.h test
+CLEANFILES+= nlm_prot_xdr.c sm_inter_xdr.c
 
-RPCSRC= ${DESTDIR}/usr/include/rpcsvc/nlm_prot.x
+RPCSRC= ${.CURDIR}/../../include/rpcsvc/nlm_prot.x
 RPCGEN= RPCGEN_CPP=${CPP:Q} rpcgen -L -C
 
+.PATH: ../rpc.statd
+
 nlm_prot_svc.c: ${RPCSRC}
 	${RPCGEN} -m -o ${.TARGET} ${RPCSRC}
 
-nlm_prot.h: ${RPCSRC}
-	${RPCGEN} -h -o ${.TARGET} ${RPCSRC}
+nlm_prot_xdr.c: ${RPCSRC}
+	${RPCGEN} -c -o ${.TARGET} ${RPCSRC}
 
 test: ${.CURDIR}/test.c
 	cc -o test ${.CURDIR}/test.c -lrpcsvc
Index: freebsd-utils-jessie/src/sbin/swapon/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/swapon/Makefile
+++ freebsd-utils-jessie/src/sbin/swapon/Makefile
@@ -9,6 +9,6 @@ MLINKS=	swapon.8 swapoff.8
 MLINKS+=swapon.8 swapctl.8
 
 DPADD=	${LIBUTIL}
-LDADD=	-lutil
+LDADD=	-lutil-freebsd -lfreebsd-glue
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/sbin/pfctl/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/pfctl/Makefile
+++ freebsd-utils-jessie/src/sbin/pfctl/Makefile
@@ -28,7 +28,7 @@ CFLAGS+= -DWITH_INET
 
 YFLAGS=
 
-LDADD+=	-lm -lmd
+LDADD+=	-lm -lbsd -lfreebsd-glue
 DPADD+=	${LIBM} ${LIBMD}
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/usr.sbin/authpf/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/authpf/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/authpf/Makefile
@@ -5,7 +5,7 @@
 PROG=	authpf
 MAN=	authpf.8
 BINOWN=	root
-BINGRP= authpf
+#BINGRP= authpf
 BINMODE= 6555
 
 SRCS=   authpf.c
@@ -13,7 +13,7 @@ SRCS=   authpf.c
 # XXX ALTQ:
 CFLAGS+= -DENABLE_ALTQ
 
-LDADD+= -lm -lmd -lutil
+LDADD+= -lm      -lutil-freebsd
 DPADD+= ${LIBM} ${LIBMD} ${LIBUTIL}
 
 WARNS?=	3
Index: freebsd-utils-jessie/src/sbin/savecore/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/savecore/Makefile
+++ freebsd-utils-jessie/src/sbin/savecore/Makefile
@@ -2,7 +2,7 @@
 
 PROG=	savecore
 DPADD=	${LIBZ}
-LDADD=	-lz
+LDADD=	-lz -lz-freebsd -lfreebsd-glue -lbsd
 MAN=	savecore.8
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/sbin/geom/class/raid/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/geom/class/raid/Makefile
+++ freebsd-utils-jessie/src/sbin/geom/class/raid/Makefile
@@ -5,6 +5,6 @@
 GEOM_CLASS=	raid
 
 DPADD=	${LIBMD}
-LDADD=	-lmd
+#LDADD=	-lmd
 
 .include <bsd.lib.mk>
Index: freebsd-utils-jessie/src/sbin/geom/class/eli/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/geom/class/eli/Makefile
+++ freebsd-utils-jessie/src/sbin/geom/class/eli/Makefile
@@ -9,7 +9,7 @@ SRCS+=	pkcs5v2.c
 SRCS+=	sha2.c
 
 DPADD=	${LIBMD} ${LIBCRYPTO}
-LDADD=	-lmd -lcrypto
+LDADD=	-lcrypto
 
 WARNS?=	3
 
Index: freebsd-utils-jessie/src/sbin/geom/class/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/geom/class/Makefile
+++ freebsd-utils-jessie/src/sbin/geom/class/Makefile
@@ -7,18 +7,14 @@ SUBDIR+=concat
 .if ${MK_OPENSSL} != "no"
 SUBDIR+=eli
 .endif
-SUBDIR+=journal
 SUBDIR+=label
 SUBDIR+=mirror
 SUBDIR+=mountver
-SUBDIR+=multipath
 SUBDIR+=nop
-SUBDIR+=part
 SUBDIR+=raid
 SUBDIR+=raid3
 SUBDIR+=sched
 SUBDIR+=shsec
 SUBDIR+=stripe
-SUBDIR+=virstor
 
 .include <bsd.subdir.mk>
Index: freebsd-utils-jessie/src/sbin/geom/class/mirror/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/geom/class/mirror/Makefile
+++ freebsd-utils-jessie/src/sbin/geom/class/mirror/Makefile
@@ -5,6 +5,6 @@
 GEOM_CLASS=	mirror
 
 DPADD=	${LIBMD}
-LDADD=	-lmd
+#LDADD=	-lmd
 
 .include <bsd.lib.mk>
Index: freebsd-utils-jessie/src/sbin/geom/class/raid3/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/geom/class/raid3/Makefile
+++ freebsd-utils-jessie/src/sbin/geom/class/raid3/Makefile
@@ -5,6 +5,6 @@
 GEOM_CLASS=	raid3
 
 DPADD=	${LIBMD}
-LDADD=	-lmd
+#LDADD=	-lmd
 
 .include <bsd.lib.mk>
Index: freebsd-utils-jessie/src/sbin/geom/core/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/geom/core/Makefile
+++ freebsd-utils-jessie/src/sbin/geom/core/Makefile
@@ -12,6 +12,6 @@ CFLAGS+= -DGEOM_CLASS_DIR=\"${GEOM_CLASS
 CFLAGS+= -I${.CURDIR}/../../../sys -I${.CURDIR} -I${.CURDIR}/..
 
 DPADD=	${LIBGEOM} ${LIBSBUF} ${LIBBSDXML} ${LIBUTIL}
-LDADD=	-lgeom -lsbuf -lbsdxml -lutil
+LDADD=	-lgeom -ldl -lbsd
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/usr.sbin/pppctl/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/pppctl/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/pppctl/Makefile
@@ -6,6 +6,6 @@ MAN=	pppctl.8
 WARNS?=	2
 
 DPADD=	${LIBPTHREAD} ${LIBEDIT} ${LIBTERMCAP}
-LDADD=	-lpthread -ledit -ltermcap
+LDADD=	-lpthread -ledit -lbsd
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/sbin/mount_nfs/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/mount_nfs/Makefile
+++ freebsd-utils-jessie/src/sbin/mount_nfs/Makefile
@@ -4,6 +4,7 @@
 
 PROG=	mount_nfs
 SRCS=	mount_nfs.c getmntopts.c mounttab.c
+LDADD=	-ltirpc -lfreebsd-glue
 MAN=	mount_nfs.8
 MLINKS=	mount_nfs.8 mount_oldnfs.8
 
Index: freebsd-utils-jessie/src/usr.sbin/mountd/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/mountd/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/mountd/Makefile
@@ -6,12 +6,12 @@ SRCS=	mountd.c getmntopts.c
 MAN=	exports.5 netgroup.5 mountd.8
 
 MOUNT=  ${.CURDIR}/../../sbin/mount
-CFLAGS+= -I${MOUNT}
+CFLAGS+= -I${MOUNT} -I${.CURDIR}/../../sys
 WARNS?= 2
 
 .PATH: ${MOUNT}
 
 DPADD=	${LIBUTIL}
-LDADD=	-lutil
+LDADD=	-ltirpc -lbsd
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/usr.sbin/arp/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/arp/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/arp/Makefile
@@ -3,6 +3,7 @@
 
 PROG=	arp
 MAN=	arp.4 arp.8
+LDADD+= -lrt -lfreebsd-glue
 
 WARNS?=	3
 
Index: freebsd-utils-jessie/src/usr.bin/netstat/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.bin/netstat/Makefile
+++ freebsd-utils-jessie/src/usr.bin/netstat/Makefile
@@ -34,7 +34,8 @@ CFLAGS+=-DPF
 BINGRP=	kmem
 BINMODE=2555
 DPADD=	${LIBKVM} ${LIBMEMSTAT} ${LIBUTIL}
-LDADD=	-lkvm -lmemstat -lutil
+LDADD=	-lkvm -lmemstat -lutil-freebsd
+LDADD+= -lrt -lfreebsd-glue
 
 .if ${MK_NETGRAPH_SUPPORT} != "no"
 SRCS+=	netgraph.c
Index: freebsd-utils-jessie/src/sbin/umount/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/umount/Makefile
+++ freebsd-utils-jessie/src/sbin/umount/Makefile
@@ -6,6 +6,8 @@ PROG=	umount
 SRCS=	umount.c vfslist.c mounttab.c
 MAN=	umount.8
 
+LDADD+=	-lfreebsd-glue
+
 MOUNT=	${.CURDIR}/../mount
 UMNTALL= ${.CURDIR}/../../usr.sbin/rpc.umntall
 CFLAGS+= -I${MOUNT} -I${UMNTALL}
Index: freebsd-utils-jessie/src/sbin/route/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/route/Makefile
+++ freebsd-utils-jessie/src/sbin/route/Makefile
@@ -9,6 +9,8 @@ SRCS=	route.c keywords.h
 WARNS?=	3
 CLEANFILES+=keywords.h
 
+LDADD+=	-lfreebsd-glue
+
 CFLAGS+= -DNS
 .if ${MK_INET_SUPPORT} != "no"
 CFLAGS+= -DINET
Index: freebsd-utils-jessie/src/sbin/mount/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/mount/Makefile
+++ freebsd-utils-jessie/src/sbin/mount/Makefile
@@ -7,6 +7,8 @@ MAN=	mount.8 mount.conf.8
 # We do NOT install the getmntopts.3 man page.
 
 DPADD=	${LIBUTIL}
-LDADD=	-lutil
+LDADD=	-lutil-freebsd
+
+LDADD+=	-lfreebsd-glue
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/usr.sbin/jail/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/jail/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/jail/Makefile
@@ -7,7 +7,9 @@ MAN=	jail.8 jail.conf.5
 SRCS=	jail.c command.c config.c state.c jailp.h jaillex.l jailparse.y y.tab.h
 
 DPADD=	${LIBJAIL} ${LIBKVM} ${LIBUTIL} ${LIBL}
-LDADD=	-ljail -lkvm -lutil -ll
+LDADD=	-ljail -lkvm -lutil-freebsd -ll
+
+LDADD+=	-lfreebsd-glue
 
 NO_WMISSING_VARIABLE_DECLARATIONS=
 
Index: freebsd-utils-jessie/src/sbin/camcontrol/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/camcontrol/Makefile
+++ freebsd-utils-jessie/src/sbin/camcontrol/Makefile
@@ -12,7 +12,7 @@ CFLAGS+= -DMINIMALISTIC
 WARNS?= 3
 .endif
 DPADD=	${LIBCAM} ${LIBSBUF} ${LIBUTIL}
-LDADD=	-lcam -lsbuf -lutil
+LDADD=	-lcam -lsbuf -lutil-freebsd
 MAN=	camcontrol.8
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/sbin/mdconfig/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/mdconfig/Makefile
+++ freebsd-utils-jessie/src/sbin/mdconfig/Makefile
@@ -6,7 +6,7 @@ PROG=		mdconfig
 MAN=		mdconfig.8
 
 DPADD=	${LIBUTIL} ${LIBGEOM} ${LIBBSDXML} ${LIBSBUF} 
-LDADD=	-lutil -lgeom -lbsdxml -lsbuf
+LDADD=	-lgeom
 
 .if ${MK_TESTS} != "no"
 SUBDIR+=    tests
Index: freebsd-utils-jessie/src/sbin/devd/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/sbin/devd/Makefile
+++ freebsd-utils-jessie/src/sbin/devd/Makefile
@@ -11,7 +11,7 @@ WARNS?=	3
 NO_SHARED?=YES
 
 DPADD=	${LIBL} ${LIBUTIL}
-LDADD=	-ll -lutil
+LDADD=	-ll
 
 LIBSTDC++ != \
 	g++ -print-file-name=crt1.o ; \
Index: freebsd-utils-jessie/src/usr.sbin/powerd/Makefile
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/powerd/Makefile
+++ freebsd-utils-jessie/src/usr.sbin/powerd/Makefile
@@ -4,6 +4,6 @@ PROG=	powerd
 MAN=	powerd.8
 
 DPADD=	${LIBUTIL}
-LDADD=	-lutil
+#LDADD=	-lutil
 
 .include <bsd.prog.mk>
Index: freebsd-utils-jessie/src/bin/Makefile
===================================================================
--- /dev/null
+++ freebsd-utils-jessie/src/bin/Makefile
@@ -0,0 +1,5 @@
+SUBDIR = \
+	kenv \
+	${NULL}
+
+.include <bsd.subdir.mk>
Index: freebsd-utils-jessie/src/sbin/Makefile
===================================================================
--- /dev/null
+++ freebsd-utils-jessie/src/sbin/Makefile
@@ -0,0 +1,36 @@
+SUBDIR = \
+	devd \
+	ifconfig \
+	route \
+	mdconfig \
+	sysctl \
+	kldload \
+	kldstat \
+	kldunload \
+	${NULL}
+
+.if !defined(RESCUE)
+SUBDIR += \
+	camcontrol \
+	ccdconfig \
+	devfs \
+	dmesg \
+	dumpon \
+	geom \
+	kldconfig \
+	mount \
+	mount_cd9660 \
+	mount_msdosfs \
+	mount_nfs \
+	mount_nullfs \
+	mount_udf \
+	mount_unionfs \
+	nfsiod \
+	pfctl \
+	savecore \
+	swapon \
+	umount \
+	${NULL}
+.endif
+
+.include <bsd.subdir.mk>
Index: freebsd-utils-jessie/src/usr.bin/Makefile
===================================================================
--- /dev/null
+++ freebsd-utils-jessie/src/usr.bin/Makefile
@@ -0,0 +1,9 @@
+SUBDIR = \
+	kdump \
+	ktrace \
+	ktrdump \
+	mkuzip \
+	netstat \
+	vmstat \
+
+.include <bsd.subdir.mk>
Index: freebsd-utils-jessie/src/usr.sbin/Makefile
===================================================================
--- /dev/null
+++ freebsd-utils-jessie/src/usr.sbin/Makefile
@@ -0,0 +1,22 @@
+SUBDIR = \
+	kbdcontrol \
+
+.if !defined(RESCUE)
+SUBDIR += \
+	acpi \
+	arp \
+	authpf \
+	jail \
+	mountd \
+	nfsd \
+	powerd \
+	ppp \
+	pppctl \
+	rpc.statd \
+	rpc.lockd \
+	rpc.umntall \
+	vidcontrol \
+	${NULL}
+.endif
+
+.include <bsd.subdir.mk>
Index: freebsd-utils-jessie/src/Makefile
===================================================================
--- /dev/null
+++ freebsd-utils-jessie/src/Makefile
@@ -0,0 +1,14 @@
+SUBDIR = \
+	bin \
+	sbin \
+	etc/devd \
+	usr.sbin \
+	share/syscons/scrnmaps \
+
+.if !defined(RESCUE)
+SUBDIR += \
+	usr.bin \
+	${NULL}
+.endif
+
+.include <bsd.subdir.mk>
