
From HEAD (SVN rev 253372)

--- a/src/sbin/mount/mount.c
+++ b/src/sbin/mount/mount.c
@@ -252,7 +252,7 @@
 	options = NULL;
 	vfslist = NULL;
 	vfstype = "ufs";
-	while ((ch = getopt(argc, argv, "adF:flo:prt:uvw")) != -1)
+	while ((ch = getopt(argc, argv, "adF:flno:prt:uvw")) != -1)
 		switch (ch) {
 		case 'a':
 			all = 1;
@@ -269,6 +269,9 @@
 		case 'l':
 			late = 1;
 			break;
+		case 'n':
+			/* For compatibility with the Linux version of mount. */
+			break;
 		case 'o':
 			if (*optarg) {
 				options = catopt(options, optarg);
--- a/src/sbin/mount/mount.8
+++ b/src/sbin/mount/mount.8
@@ -111,6 +111,9 @@
 .Fl a
 option, also mount those file systems which are marked as
 .Dq Li late .
+.It Fl n
+For compatibility with some other implementations, this flag is
+currently a no-op.
 .It Fl o
 Options are specified with a
 .Fl o
