Index: freebsd-utils-jessie/src/usr.sbin/ppp/bundle.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/bundle.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/bundle.c
@@ -699,7 +699,7 @@ bundle_Create(const char *prefix, int ty
   static struct bundle bundle;		/* there can be only one */
   int enoentcount, err, minunit, maxunit;
   const char *ifname;
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
   int kldtried;
 #endif
 #if defined(TUNSIFMODE) || defined(TUNSLMODE) || defined(TUNSIFHEAD)
@@ -720,7 +720,7 @@ bundle_Create(const char *prefix, int ty
   }
   err = ENOENT;
   enoentcount = 0;
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
   kldtried = 0;
 #endif
   for (bundle.unit = minunit; bundle.unit != maxunit; bundle.unit++) {
@@ -730,7 +730,7 @@ bundle_Create(const char *prefix, int ty
     if (bundle.dev.fd >= 0)
       break;
     else if (errno == ENXIO || errno == ENOENT) {
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
       if (bundle.unit == minunit && !kldtried++) {
         /*
          * Attempt to load the tunnel interface KLD if it isn't loaded
Index: freebsd-utils-jessie/src/usr.sbin/ppp/cbcp.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/cbcp.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/cbcp.c
@@ -28,7 +28,7 @@
 
 #include <sys/param.h>
 
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #include <netinet/in.h>
 #endif
 #include <sys/un.h>
Index: freebsd-utils-jessie/src/usr.sbin/ppp/chap_ms.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/chap_ms.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/chap_ms.c
@@ -29,8 +29,9 @@
  * $FreeBSD: stable/10/usr.sbin/ppp/chap_ms.c 134789 2004-09-05 01:46:52Z brian $
  */
 
+#include <sys/param.h>
 #include <ctype.h>
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #include <openssl/des.h>
 #include <sha.h>
 #else
@@ -187,7 +188,7 @@ GenerateNTResponse(char *AuthenticatorCh
   ChallengeResponse(Challenge, PasswordHash, Response);
 }
 
-#ifndef __FreeBSD__
+#ifndef __FreeBSD_kernel__
 #define LENGTH 20
 static char *
 SHA1_End(SHA_CTX *ctx, char *buf)
Index: freebsd-utils-jessie/src/usr.sbin/ppp/defs.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/defs.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/defs.c
@@ -39,16 +39,16 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
 #include <sys/module.h>
 #endif
 #include <termios.h>
-#ifndef __FreeBSD__
+#ifndef __FreeBSD_kernel__
 #include <time.h>
 #endif
 #include <unistd.h>
 
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
 #include "id.h"
 #include "log.h"
 #endif
@@ -411,7 +411,7 @@ Concatinate(char *buf, size_t sz, int ar
   }
 }
 
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
 int
 loadmodules(int how, const char *module, ...)
 {
Index: freebsd-utils-jessie/src/usr.sbin/ppp/defs.h
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/defs.h
+++ freebsd-utils-jessie/src/usr.sbin/ppp/defs.h
@@ -28,8 +28,10 @@
  * $FreeBSD: stable/10/usr.sbin/ppp/defs.h 252826 2013-07-05 19:26:01Z rmh $
  */
 
+#include <sys/param.h>
+
 /* Check the following definitions for your machine environment */
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 # define  MODEM_LIST	"/dev/cuau1\0/dev/cuau0"	/* name of tty device */
 #else
 # ifdef __OpenBSD__
Index: freebsd-utils-jessie/src/usr.sbin/ppp/id.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/id.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/id.c
@@ -40,7 +40,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <sysexits.h>
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
 #include <sys/linker.h>
 #endif
 #include <unistd.h>
@@ -262,7 +262,7 @@ ID0kill(pid_t pid, int sig)
   return result;
 }
 
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
 int
 ID0kldload(const char *dev)
 {
Index: freebsd-utils-jessie/src/usr.sbin/ppp/id.h
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/id.h
+++ freebsd-utils-jessie/src/usr.sbin/ppp/id.h
@@ -26,6 +26,8 @@
  * $FreeBSD: stable/10/usr.sbin/ppp/id.h 202192 2010-01-13 17:54:32Z ed $
  */
 
+#include <sys/param.h>
+
 #ifndef NOSUID
 struct utmpx;
 struct sockaddr_un;
@@ -46,7 +48,7 @@ extern void ID0logout(const struct utmpx
 extern int ID0bind_un(int, const struct sockaddr_un *);
 extern int ID0connect_un(int, const struct sockaddr_un *);
 extern int ID0kill(pid_t, int);
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
 extern int ID0kldload(const char *);
 #endif
 #ifndef NONETGRAPH
@@ -70,7 +72,7 @@ extern int ID0NgMkSockNode(const char *,
 #define ID0connect_un(s, n) \
 	connect(s, (const struct sockaddr *)(n), sizeof *(n))
 #define ID0kill kill
-#if defined(__FreeBSD__) && !defined(NOKLDLOAD)
+#if defined(__FreeBSD_kernel__) && !defined(NOKLDLOAD)
 #include <sys/param.h>
 #include <sys/linker.h>
 #define ID0kldload kldload
Index: freebsd-utils-jessie/src/usr.sbin/ppp/iface.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/iface.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/iface.c
@@ -31,7 +31,7 @@
 #include <netinet/in.h>
 #include <net/if.h>
 #include <net/if_dl.h>
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #include <net/if_var.h>
 #endif
 #include <net/route.h>
@@ -666,7 +666,7 @@ iface_ChangeFlags(const char *ifname, in
     close(s);
     return 0;
   }
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
   new_flags = (ifrq.ifr_flags & 0xffff) | (ifrq.ifr_flagshigh << 16);
 #else
   new_flags = ifrq.ifr_flags & 0xffff;
@@ -677,7 +677,7 @@ iface_ChangeFlags(const char *ifname, in
   else
     new_flags &= ~flags;
   ifrq.ifr_flags = new_flags & 0xffff;
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
   ifrq.ifr_flagshigh = new_flags >> 16;
 #endif
 
Index: freebsd-utils-jessie/src/usr.sbin/ppp/lqr.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/lqr.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/lqr.c
@@ -30,7 +30,7 @@
 
 #include <sys/param.h>
 
-#ifdef __FreeBSD__
+#ifdef __FreeBSD_kernel__
 #include <netinet/in.h>
 #endif
 #include <sys/un.h>
Index: freebsd-utils-jessie/src/usr.sbin/ppp/tun.c
===================================================================
--- freebsd-utils-jessie.orig/src/usr.sbin/ppp/tun.c
+++ freebsd-utils-jessie/src/usr.sbin/ppp/tun.c
@@ -29,7 +29,7 @@
 #include <sys/param.h>
 
 #include <sys/socket.h>		/* For IFF_ defines */
-#ifndef __FreeBSD__
+#ifndef __FreeBSD_kernel__
 #include <net/if.h>		/* For IFF_ defines */
 #endif
 #include <net/route.h>
