--- a/src/usr.sbin/rpc.lockd/lockd.c
+++ b/src/usr.sbin/rpc.lockd/lockd.c
@@ -905,6 +905,7 @@
 						sin->sin_family = AF_INET;
 						sin->sin_port = htons(0);
 						sin->sin_addr.s_addr = htonl(INADDR_ANY);
+						sin->sin_len = sizeof(struct sockaddr_in);
 						res->ai_addr = (struct sockaddr*) sin;
 						res->ai_addrlen = (socklen_t)
 						    sizeof(res->ai_addr);
@@ -916,6 +917,7 @@
 						sin6->sin6_family = AF_INET6;
 						sin6->sin6_port = htons(0);
 						sin6->sin6_addr = in6addr_any;
+						sin6->sin6_len = sizeof(struct sockaddr_in6);
 						res->ai_addr = (struct sockaddr*) sin6;
 						res->ai_addrlen = (socklen_t) sizeof(res->ai_addr);
 						break;
--- a/src/usr.bin/vmstat/Makefile
+++ b/src/usr.bin/vmstat/Makefile
@@ -4,7 +4,7 @@
 PROG=	vmstat
 MAN=	vmstat.8
 DPADD=	${LIBDEVSTAT} ${LIBKVM} ${LIBMEMSTAT} ${LIBUTIL}
-LDADD=	-ldevstat -lkvm -lmemstat -lutil
+LDADD=	-ldevstat -lkvm -lmemstat -lutil-freebsd
 
 WARNS?=	1
 
